elastic/kibana
# [Ingest Manager ] prepend kibana asset ids with package name

https://github.com/elastic/kibana/issues/65035

The kibana assets installed by the Ingest Manager could conflict with ones with the same ids installed by Beats.  This change will keep separate versions by modifying the ids of the assets in the Ingest Manager, as well as the references to ids in Dashboard assets, before installing

## Changes

- prepends the `id` of each installed kibana asset with the package name 
- prepends the` id` of each asset the asset references (in the asset's `references` property)
- only asset ids of type` dashboard`, `visualization`,` search`, and `map` are modified

## Test

- restart ES
- resfresh kibana in the browser
- view saved objects of any assets and see that ones belong to the package installed (by default, system and endpoint) are prepended with the package name, eg: http://localhost:5601/ssg/api/saved_objects/_find?fields=type&type=dashboard).  
- view saved objects of `epm-package` and see it references the correct modified ids (eg http://localhost:5601/ssg/api/saved_objects/_find?fields=installed&type=epm-packages)
- In Kibana Navigate to Dashboards, Visualize, Maps, or Search and there should be a list of the assets with no errors.  Clicking on the assets should appear and be reflected in the browser url:
<img width="813" alt="Screen Shot 2020-07-01 at 3 17 19 PM" src="https://user-images.githubusercontent.com/1676003/86282851-fd33fb00-bbad-11ea-9b45-74672c04906c.png">
- install a package and repeat
